binlog_writer.go

```diff
@@ -110,7 +116,7 @@ func (b *BinlogWriter) writeEvents(events []DMLEvent) error {
 		return fmt.Errorf("exec query (%d bytes): %v", len(query), err)
 	}
 
-	// lastWrittenBinlogPosition = events[len(events)-1].BinlogPosition()
+	b.StateTracker.UpdateLastWrittenBinlogPosition(events[len(events)-1].BinlogPosition())
```

Comment:
is there a test that would fail if we removed this line?

how about one that would fail if it was executed at the wrong time?

Reply:
I thought about this for quite a bit and I'm not sure if there is a way to positively assert this line: we can't know what the binlog position is when we execute a statement easily. To do so we need to tail the binlogs in the tests, which is challenging. The best way I can think of testing this is blackbox testing: if we resume a dumped state, does it correctly copy the table with writes happen simultaneously. This should be enough that the dumped coordinates are "correct".

Reply:
Actually I think there's a way to assert the dumped binlog position from the ruby test, but I'll have to spend sometime looking at it. Either way, I don't think I can do it from this PR.